### DISCUSSION TITLE
Help with an error when re-opening a Wizard

### DISCUSSION THREAD

**Discussion by EddRobinson** [tui-cs/Terminal.Gui#2374] [Q&A] [answered]
`Private Static void Main()`
`{`
`Application.Init();`
`// Setting up of the Wizard`
`}`

I call / show the wizard by using `Application.Run(setupWizard);`
I have tried closing it with the `setupWizard.Finished`  button and by using `Application.RequestStop(setupWizard);`

They both seem to work fine. But if I try and run the Wizard again at any point, I get the following error

![image](https://user-images.githubusercontent.com/111066518/220392883-f3e91886-f06f-4215-9cf4-33b9b23df999.png)

Any idea's what I could've done wrong?

**Comment by tznind** [comment, marked as answer]
When you exit a `Dialog` it's `Dispose` method is called which means it cannot be reused to launch again.  I think you would need to create a new instance and show that one instead.

**Reply by EddRobinson** [reply]
> When you exit a `Dialog` it's `Dispose` method is called which means it cannot be reused to launch again. I think you would need to create a new instance and show that one instead.

Ahh yeah, many thanks. Sometimes the simplest of things still gets me

**Reply by tznind** [reply]
No worries, the error message isn't the clearest :)  glad you got it fixed.
